qiskit_experiments/framework/composite/composite_experiment_data.py

```diff
@@ -17,40 +17,50 @@
 from qiskit.result import marginal_counts
 from qiskit.exceptions import QiskitError
 from qiskit_experiments.framework.experiment_data import ExperimentData
+from qiskit_experiments.database_service import DbExperimentDataV1, DatabaseServiceV1
 
 
 class CompositeExperimentData(ExperimentData):
     """Composite experiment data class"""
 
-    def __init__(
-        self,
-        experiment,
-        backend=None,
-        job_ids=None,
-    ):
+    def __init__(self, experiment, backend=None, job_ids=None, parent_id=None, root_id=None):
         """Initialize experiment data.
 
         Args:
             experiment (CompositeExperiment): experiment object that generated the data.
             backend (Backend): Optional, Backend the experiment runs on. It can either be a
                 :class:`~qiskit.providers.Backend` instance or just backend name.
             job_ids (list[str]): Optional, IDs of jobs submitted for the experiment.
-
-        Raises:
-            ExperimentError: If an input argument is invalid.
+            parent_id (str): Optional, ID of the parent experiment data in a composite experiment
+            root_id (str): Optional, ID of the root experiment data in a composite experiment
         """
 
         super().__init__(
-            experiment,
-            backend=backend,
-            job_ids=job_ids,
+            experiment, backend=backend, job_ids=job_ids, parent_id=parent_id, root_id=root_id
         )
 
+        # In a composite setting, an experiment is tagged with its direct parent and with the root.
+        # This is done in the ExperimentData constructor, except for the root experiment,
+        # for whom this is done here
+        root_id = root_id if root_id is not None else self.experiment_id
+        if len(self.tags) == 0:
+            self.tags = ["root exp id: " + root_id]
```

Comment:
Can we tag both root and parent experiments?

Reply:
This is the root here, it has no parent

Reply:
Never mind, I misunderstood